web/src/pages/Route/components/DataLoader/Import.tsx

```diff
@@ -0,0 +1,262 @@
+                      required: true,
+                      message: formatMessage({ id: 'page.route.data_loader.tips.select_type' }),
+                    },
+                  ]}
+                  initialValue={importType}
+                >
+                  <Select onChange={(value: ImportType) => setImportType(value)}>
+                    <Select.Option value="openapi3">
+                      {formatMessage({ id: 'page.route.data_loader.types.openapi3' })}
+                    </Select.Option>
+                    <Select.Option value="openapi_legacy" disabled>
+                      {formatMessage({ id: 'page.route.data_loader.types.openapi_legacy' })}
+                    </Select.Option>
+                  </Select>
+                </Form.Item>
+              </Col>
+              <Col span={12}>
+                <Form.Item
+                  name="task_name"
+                  label={formatMessage({ id: 'page.route.data_loader.labels.task_name' })}
+                  rules={[
+                    {
+                      required: true,
+                      message: formatMessage({ id: 'page.route.data_loader.tips.input_task_name' }),
+                    },
+                  ]}
+                >
+                  <Input
+                    placeholder={formatMessage({
+                      id: 'page.route.data_loader.tips.input_task_name',
+                    })}
+                  />
+                </Form.Item>
+              </Col>
+            </Row>
+            <Option type={importType}></Option>
+            <Divider />
+            <Row gutter={16}>
+              <Col span={24}>
+                <Form.Item label={formatMessage({ id: 'page.route.data_loader.labels.upload' })}>
+                  <Upload
+                    fileList={uploadFileList as any}
+                    beforeUpload={(file) => {
+                      setUploadFileList([file]);
+                      return false;
+                    }}
+                    onRemove={() => {
+                      setUploadFileList([]);
+                    }}
+                  >
+                    <Button icon={<UploadOutlined />}>
+                      {formatMessage({ id: 'page.route.data_loader.tips.click_upload' })}
+                    </Button>
+                  </Upload>
+                </Form.Item>
+              </Col>
+            </Row>
+          </Form>
+        )}
+        {state === 'result' && (
+          <Result
+            status={importResult.success ? 'success' : 'error'}
+            title={`${formatMessage({ id: 'page.route.data_loader.import' })} ${
+              importResult.success
+                ? formatMessage({ id: 'component.status.success' })
+                : formatMessage({ id: 'component.status.fail' })
+            }`}
+            extra={[
+              <Button
+                type="primary"
+                onClick={() => {
+                  setState('import');
+                  onClose?.();
```

Comment:
Would it be better to change the onClose method to a mandatory parameter?

Reply:
indeed

Reply:
improved